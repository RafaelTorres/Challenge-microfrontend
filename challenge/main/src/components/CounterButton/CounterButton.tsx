import { FC } from 'react';
import styles from './styles.module.css';

export interface CounterButtonProps {
  onClick: () => void;
  label?: string;
}

const CounterButton: FC<CounterButtonProps> = ({ onClick, label = 'Add to Cart' }) => (
  <button onClick={onClick} type="button" className={styles.button}>
    {label}
  </button>
);

export default CounterButton;
