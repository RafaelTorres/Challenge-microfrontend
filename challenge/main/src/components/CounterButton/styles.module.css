.button {
  width: 4rem;
  height: 4rem;
  border-radius: var(--border-radius);
  border: 1px solid var(--button-border-color);
  background-color: var(--button-background);
  color: var(--button-color);
  font-size: 4rem;
  line-height: 0;
  cursor: pointer;
}

.button:active {
  background-color: var(--button-backgroun-feedback);
}
