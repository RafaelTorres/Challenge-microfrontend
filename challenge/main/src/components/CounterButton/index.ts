import CounterButton, { CounterButtonProps } from './CounterButton';

export type { CounterButtonProps };

export default CounterButton;
