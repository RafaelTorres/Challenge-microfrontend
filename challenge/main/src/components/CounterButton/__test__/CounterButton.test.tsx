import { fireEvent, render, screen } from '@testing-library/react';
import CounterButton, { CounterButtonProps } from '..';

const handleClick = jest.fn();
const defaultProps = {
  onClick: handleClick,
};

const renderPage = (props?: Partial<CounterButtonProps>) => render(<CounterButton {...defaultProps} {...props} />);

describe('Counter Button', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the component', () => {
    const { container } = renderPage();
    const button = screen.getByText(/Add to Cart/);
    expect(button).toBeInTheDocument();
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalled();
    expect(handleClick).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });

  it('should render the component with custom label', () => {
    const { container } = renderPage({ label: 'Agregar item' });
    const button = screen.queryByText(/Agregar item/);
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
