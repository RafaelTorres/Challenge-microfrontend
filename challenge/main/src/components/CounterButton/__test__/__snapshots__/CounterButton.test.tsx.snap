// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Button should render the component 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Add to Cart
  </button>
</div>
`;

exports[`Counter Button should render the component with custom label 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Agregar item
  </button>
</div>
`;
